src/ReactTelephoneInput.js

```diff
@@ -336,6 +333,19 @@ var ReactTelephoneInput = React.createClass({
 
         this._fillDialCode();
     },
+    _mapPropsToState(props) {
+        var inputNumber = props.initialValue || props.value || '';
```

Comment:
can you add check..So we change inputNumber only when we detect change, or else we will changes we have made on state something like !_.isEqual(nextProps, this.props) {//code here}